core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -209,12 +209,11 @@ public void deleteFileAsync(int fileId) {
       return aggregateCompletionStage.freeze().thenApply(AtomicLong::get);
    }
 
-   public long approximateSize() {
+   public long approximateSize(IntSet filterSegments) {
       long size = 0;
-      for (Segment seg : segments) {
-         size += seg.size.get();
-         if (size < 0) {
-            return Long.MAX_VALUE;
+      for (int s = 0; s < segments.length; s++) {
```

Comment:
Unfortunately, this segment is not the same as a cache segment. The SIFS uses its own notion of segment to reduce B-tree overhead by splitting it into its own "segments" keyed by the hashCode.

We probably need to just do the same calculation that PersistenceMangerImpl does when segmented.

Reply:
Thanks, I will also add a test in `BaseNonBlockingStoreTest`